### PR TITLE
Upgrade to flutter 3.3.0 + lint upgrade and fixes

### PR DESCRIPTION
**Pull request by WieFel**
also moved lint dependency to `dev_dependencies`, as I couldn't find any reason why it was listed under normal dependencies

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## [Unreleased]
```
Comment (calvintam236) [inline CHANGELOG.md:3]: Since this is a breaking change, this should be `3.0.0`

**Comment by calvintam236** [comment]
https://github.com/TinyCommunity/tinycolor2/blob/6670336b21b74bd528a4a0881cd94ce610ef2ec6/pubspec.yaml#L7 should update to `>=3.3.0 <4.0.0` as this blocks on flutter v3 or above and `lint` upgrade will break older Flutter versions.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+
+- Upgrade to Flutter 3.3.0
+
```
Comment (calvintam236) [inline CHANGELOG.md:6]: missing `lint` upgrade

**Review by calvintam236 — CHANGES_REQUESTED** [review]
See converstation.

File: CHANGELOG.md (hunk 1 above)
Comment (WieFel) [reply CHANGELOG.md:3]: Ok for me

File: CHANGELOG.md (hunk 2 above)
Comment (WieFel) [reply CHANGELOG.md:6]: will add...

**Comment by calvintam236** [comment]
> https://github.com/TinyCommunity/tinycolor2/blob/6670336b21b74bd528a4a0881cd94ce610ef2ec6/pubspec.yaml#L7
> 
> should update to `>=3.3.0 <4.0.0` as this blocks on flutter v3 or above and `lint` upgrade will break older Flutter versions.

@WieFel don’t forget this

**Comment by WieFel** [comment]
> https://github.com/TinyCommunity/tinycolor2/blob/6670336b21b74bd528a4a0881cd94ce610ef2ec6/pubspec.yaml#L7
> should update to `>=3.3.0 <4.0.0` as this blocks on flutter v3 or above and `lint` upgrade will break older Flutter versions.

I think putting `>=3.0.0` suffices.
See
image
Lint 1.10.0 requires dart `>=2.17.0` which corresponds to Flutter 3.0.0

**Comment by WieFel** [comment]
Just noticed that setting `sdk: ">=3.0.0 <4.0.0" ` doesn't make sense, as the sdk specifies the dart version.
Dart 3.0.0 doesn't even exist yet.

So now I set `sdk: ">=2.17.0 <3.0.0" ` (corresponds to flutter 3.0.0) and `flutter: ">=3.0.0"`

**Review by calvintam236 — APPROVED** [review]

**Comment by calvintam236** [comment]
@WieFel you can now make a new release

**Comment by calvintam236** [comment]
@WieFel oops, i think you forgot to update this two lines

https://github.com/TinyCommunity/tinycolor2/blob/116ffe559e2cc6cfd2d3feff1f0aacd186e120ae/example/pubspec.yaml#L21-L22

**Comment by WieFel** [comment]
I did update it. Check on master:
https://github.com/TinyCommunity/tinycolor2/blob/116ffe559e2cc6cfd2d3feff1f0aacd186e120ae/pubspec.yaml#L7-L8

**Comment by calvintam236** [comment]
> I did update it. Check on master:
> 
> https://github.com/TinyCommunity/tinycolor2/blob/116ffe559e2cc6cfd2d3feff1f0aacd186e120ae/pubspec.yaml#L7-L8

Not the package one. I’m referring to the `example/` one.

**Comment by WieFel** [comment]
#64

**Pull request merged**